src/informer.ts

```diff
@@ -0,0 +1,24 @@
+import { ListWatch } from './cache';
+import { KubeConfig } from './config';
+import { KubernetesObject } from './types';
+import { Watch } from './watch';
+
+export type ObjectCallback<T extends KubernetesObject> = (obj: T) => void;
+export type ListCallback<T extends KubernetesObject> = (list: T[], ResourceVersion: string) => void;
+
+export const ADD: string = 'add';
```

Comment:
Would this not be a good use of [enums](https://www.typescriptlang.org/docs/handbook/enums.html) might make the validation code a bit easier too?

Reply:
I feel like using enums in a `on(...)` clause for node is not typical, but I'm open to being wrong.

Reply:
You are right just thought it would clean up the `if checking` thought this could work interchangeably but clearly, typescript enums don't work like that.